File: SDKs/yodlee.com/1.1.0/python/src/sdk/models/shared/createaccountinfo.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import accountaddress as shared_accountaddress
from ..shared import money as shared_money

class CreateAccountInfoFrequencyEnum(str, Enum):
    DAILY = "DAILY"
    ONE_TIME = "ONE_TIME"
    WEEKLY = "WEEKLY"
    EVERY_2_WEEKS = "EVERY_2_WEEKS"
    SEMI_MONTHLY = "SEMI_MONTHLY"
    MONTHLY = "MONTHLY"
    QUARTERLY = "QUARTERLY"
    SEMI_ANNUALLY = "SEMI_ANNUALLY"
    ANNUALLY = "ANNUALLY"
    EVERY_2_MONTHS = "EVERY_2_MONTHS"
    EBILL = "EBILL"
    FIRST_DAY_MONTHLY = "FIRST_DAY_MONTHLY"
    LAST_DAY_MONTHLY = "LAST_DAY_MONTHLY"
    EVERY_4_WEEKS = "EVERY_4_WEEKS"
    UNKNOWN = "UNKNOWN"
    OTHER = "OTHER"

class CreateAccountInfoValuationTypeEnum(str, Enum):
    SYSTEM = "SYSTEM"
    MANUAL = "MANUAL"


@dataclass_json
@dataclasses.dataclass
class CreateAccountInfo:
    account_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('accountName') }})
    account_type: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('accountType') }})
    account_number: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accountNumber') }})
    address: Optional[shared_accountaddress.AccountAddress] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('address') }})
    amount_due: Optional[shared_money.Money] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('amountDue') }})
    balance: Optional[shared_money.Money] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('balance') }})
    due_date: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dueDate') }})
    frequency: Optional[CreateAccountInfoFrequencyEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('frequency') }})
    home_value: Optional[shared_money.Money] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('homeValue') }})
    include_in_net_worth: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('includeInNetWorth') }})
    memo: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('memo') }})
    nickname: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('nickname') }})
    valuation_type: Optional[CreateAccountInfoValuationTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('valuationType') }})
    

File: SDKs/yodlee.com/1.1.0/typescript/src/sdk/models/shared/createaccountinfo.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { AccountAddress } from "./accountaddress";
import { Money } from "./money";


export enum CreateAccountInfoFrequencyEnum {
    Daily = "DAILY",
    OneTime = "ONE_TIME",
    Weekly = "WEEKLY",
    Every2Weeks = "EVERY_2_WEEKS",
    SemiMonthly = "SEMI_MONTHLY",
    Monthly = "MONTHLY",
    Quarterly = "QUARTERLY",
    SemiAnnually = "SEMI_ANNUALLY",
    Annually = "ANNUALLY",
    Every2Months = "EVERY_2_MONTHS",
    Ebill = "EBILL",
    FirstDayMonthly = "FIRST_DAY_MONTHLY",
    LastDayMonthly = "LAST_DAY_MONTHLY",
    Every4Weeks = "EVERY_4_WEEKS",
    Unknown = "UNKNOWN",
    Other = "OTHER"
}

export enum CreateAccountInfoValuationTypeEnum {
    System = "SYSTEM",
    Manual = "MANUAL"
}


export class CreateAccountInfo extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=accountName" })
  accountName: string;

  @SpeakeasyMetadata({ data: "json, name=accountNumber" })
  accountNumber?: string;

  @SpeakeasyMetadata({ data: "json, name=accountType" })
  accountType: string;

  @SpeakeasyMetadata({ data: "json, name=address" })
  address?: AccountAddress;

  @SpeakeasyMetadata({ data: "json, name=amountDue" })
  amountDue?: Money;

  @SpeakeasyMetadata({ data: "json, name=balance" })
  balance?: Money;

  @SpeakeasyMetadata({ data: "json, name=dueDate" })
  dueDate?: string;

  @SpeakeasyMetadata({ data: "json, name=frequency" })
  frequency?: CreateAccountInfoFrequencyEnum;

  @SpeakeasyMetadata({ data: "json, name=homeValue" })
  homeValue?: Money;

  @SpeakeasyMetadata({ data: "json, name=includeInNetWorth" })
  includeInNetWorth?: string;

  @SpeakeasyMetadata({ data: "json, name=memo" })
  memo?: string;

  @SpeakeasyMetadata({ data: "json, name=nickname" })
  nickname?: string;

  @SpeakeasyMetadata({ data: "json, name=valuationType" })
  valuationType?: CreateAccountInfoValuationTypeEnum;
}
